src/java.base/share/classes/java/lang/foreign/Arena.java

```diff
@@ -0,0 +1,170 @@
+ * <p>
+ * Shared memory sessions, on the other hand, have no owner thread. The segments created by a shared arena
+ * can be {@linkplain MemorySession#isAccessibleBy(Thread) accessed} by multiple threads. This might be useful when
+ * multiple threads need to access the same memory segment concurrently (e.g. in the case of parallel processing).
+ * Moreover, a shared arena {@linkplain #isCloseableBy(Thread) can be closed} by any thread.
+ *
+ * @since 20
+ */
+@PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
+public interface Arena extends SegmentAllocator, AutoCloseable {
+
+    /**
+     * Creates a native memory segment with the given size (in bytes) and alignment constraint (in bytes).
+     * The returned segment is associated with the arena's memory session.
+     * The segment's {@link MemorySegment#address() address} is the starting address of the
+     * allocated off-heap memory region backing the segment, and the address is 
+     * aligned according the provided alignment constraint.
+     *
+     * @implSpec
+     * The default implementation of this method is equivalent to the following code:
+     * {@snippet lang=java :
+     * MemorySegment.allocateNative(bytesSize, byteAlignment, session());
+     * }
+     *
+     * @param byteSize the size (in bytes) of the off-heap memory block backing the native memory segment.
+     * @param byteAlignment the alignment constraint (in bytes) of the off-heap region of memory backing the native memory segment.
+     * @return a new native memory segment.
+     * @throws IllegalArgumentException if {@code bytesSize < 0}, {@code alignmentBytes <= 0}, or if {@code alignmentBytes}
+     * is not a power of 2.
+     * @throws IllegalStateException if the session associated with this arena is not {@linkplain MemorySession#isAlive() alive}.
+     * @throws WrongThreadException if this method is called from a thread {@code T},
+     * such that {@code session.isAccessibleBy(T) == false}.
+     * @see MemorySegment#allocateNative(long, long, MemorySession)
+     */
+    @Override
+    default MemorySegment allocate(long byteSize, long byteAlignment) {
+        return MemorySegment.allocateNative(byteSize, byteAlignment, session());
+    }
+
+    /**
+     * {@return the session associated with this arena}
+     */
+    MemorySession session();
+
+    /**
+     * Closes this arena. If this method completes normally, the arena session becomes not {@linkplain MemorySession#isAlive() alive},
+     * and all the memory segments associated with it can no longer be accessed. Furthermore, any off-heap region of memory backing the
+     * segments associated with that memory session are also released.
+     * @throws IllegalStateException if the session associated with this arena is not {@linkplain MemorySession#isAlive() alive}.
+     * @throws IllegalStateException if this session is {@linkplain MemorySession#whileAlive(Runnable) kept alive} by another client.
```

Comment:
```suggestion
     * @throws IllegalStateException if the arena's session is not {@linkplain MemorySession#isAlive() alive}.
     * @throws IllegalStateException if the arena's session is {@linkplain MemorySession#whileAlive(Runnable) kept alive}.
```

Note i removed "by another client". I wanted to say "by another thread", but then there is the case of calling close from within the Runnable passed to whileAlive, so i wanted to say "by another caller". But, i think this can all be implied and we don't need to say anything.